notifications/gotify.go

```diff
@@ -0,0 +1,99 @@
+package notifications
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	gotifyType = "gotify"
+)
+
+type gotifyTypeNotifier struct {
+	gotifyURL      string
+	gotifyAppToken string
+	logLevels      []log.Level
+}
+
+func newGotifyNotifier(c *cobra.Command, acceptedLogLevels []log.Level) typeNotifier {
+	flags := c.PersistentFlags()
+
+	gotifyURL, _ := flags.GetString("notification-gotify-url")
+	if len(gotifyURL) < 1 {
+		log.Fatal("Required argument --notification-gotify-url(cli) or WATCHTOWER_NOTIFICATION_GOTIFY_URL(env) is empty.")
+	} else if !(strings.HasPrefix(gotifyURL, "http://") || strings.HasPrefix(gotifyURL, "https://")) {
+		log.Fatal("Gotify URL must start with \"http://\" or \"https://\"")
+	} else if strings.HasPrefix(gotifyURL, "http://") {
+		log.Warn("Using an HTTP url fpr Gotify is insecure")
+	}
+
+	gotifyToken, _ := flags.GetString("notification-gotify-token")
+	if len(gotifyToken) < 1 {
+		log.Fatal("Required argument --notification-gotify-token(cli) or WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN(env) is empty.")
+	}
+
+	n := &gotifyTypeNotifier{
+		gotifyURL:      gotifyURL,
+		gotifyAppToken: gotifyToken,
+		logLevels:      acceptedLogLevels,
+	}
+
+	log.AddHook(n)
+
+	return n
+}
+
+func (n *gotifyTypeNotifier) StartNotification() {}
+
+func (n *gotifyTypeNotifier) SendNotification() {}
+
+func (n *gotifyTypeNotifier) Levels() []log.Level {
+	return n.logLevels
+}
+
+func (n *gotifyTypeNotifier) getURL() string {
+	url := n.gotifyURL
+	if !strings.HasSuffix(url, "/") {
+		url += "/"
+	}
+	return url + "message?token=" + n.gotifyAppToken
+}
+
+func (n *gotifyTypeNotifier) Fire(entry *log.Entry) error {
+
+	go func() {
+		jsonBody, err := json.Marshal(gotifyMessage{
+			Message:  "(" + entry.Level.String() + "): " + entry.Message,
+			Title:    "WatchTower",
+			Priority: 0,
+		})
+		if err != nil {
+			fmt.Println("Failed to create JSON body for Gotify notification: ", err)
+			return
+		}
+
+		resp, err := http.Post(n.getURL(), "application/json", bytes.NewBuffer([]byte(jsonBody)))
```

Comment:
I'd prefer if you'd create the buffer before using it with the http client to improve readability.